### DISCUSSION TITLE
Add Additional Options to Image Crop Tool

### DISCUSSION THREAD

**Discussion by michaelshowes** [payloadcms/payload#3557] [Feature Requests & Ideas]
I'd like to see the ability to lock aspect ratio or restrict crop size with the image crop tool. Essentially the same functionality as the [Aspect Ratio Image Crop](https://github.com/joppuyo/acf-image-aspect-ratio-crop) plugin for Advanced Custom Fields. The option to lock the aspect ratio or use a specific pixel size may look something like:

```ts
upload: {
  cropOptions: {
    mode: 'aspect-ratio' | 'fixed-size' | 'free-crop',
    aspectRatio: 16:9 | 1:1 | etc,  // for aspect-ratio
    fixedWidth: '800',  // for fixed size
    fixedHeight: '500'  // for fixed size
  }
}
```

Something along those lines.

**Comment by revnelson** [comment]
Aspect ratio options would be amaaaazing!

**Comment by HarleySalas** [comment]
This is something that should be implemented, unquestionably.

Especially now that Payload 3.0 is built with `Next`, `aspect-ratio` should be one of the most widely-used features needed, to work with `next/image`. When using `next/image`, there is no point in upscaling, or downscaling uploaded images. 

Even if you have an image that you know will be 300 pixels wide, you should **not** be including an `imageSize` width of 300. [Device Pixel Ratios](https://cloudinary.com/glossary/device-pixel-ratio#:~:text=Device%20Pixel%20Ratio%20(DPR)%20is,(depending%20on%20your%20settings).) are a factor.

Images should never be scaled up. It doesn't make sense to input **any** dimensions in to a payload `imageSize`, as you restrict your image from being able to provide applicable image sizes for devices with high device pixel ratios. It's nice to have these features that currently exist, but the current upload features are lacking for real-world implementations.

I am absolutely adamant that 99% of Nextjs users should be simply specifying a focal point, aspect ratio and the resulting images should be the largest possible versions of the respective aspect ratios provided, taking the focal point in to consideration. (of course cropping is nice too, but that is a separate thing) Payload is not an image service and it shouldn't be-- At least give it the tools so that we can format our images and allow `next/image` to handle the `sizes` for different screens.

**Comment by appinventiv-abhijeet-singh1047** [comment]
Any update on this ?

**Comment by michaelshowes** [comment]
Yeah, this would still be immensely helpful.

**Comment by FloerlPhilip** [comment]
This would be an amazing addition

**Comment by BedroomDev** [comment]
+1 Im highly interested in this as well!

**Comment by hightechhorn** [comment]
+1 This would be very helpful in my projects as well. looking forward to this update

**Comment by WingRa7** [comment]
Yes i would love to see, with a minimum crop size to stop the other image sizes breaking.
A focal point crop preview would be the next thing for me.

**Comment by stefan-golus** [comment]
up

**Comment by swheeler7** [comment]
+1 this would be great to have. Would love to be able to lock the aspect ratio (with specific presets defined in the config) while cropping

**Comment by bohdanchornopolskyi** [comment]
+1, my team misses this feature too, aspect ratio while cropping
